### PR TITLE
Client app account info

### PR DESCRIPTION
**Pull request by ArinaSavchenko**

### CODE REVIEW

File: Airline_Web_API/Airline_Web_API/Services/JwtService.cs [hunk 1]
```diff
@@ -42,10 +42,10 @@ public string GenerateJwtToken(User user)
         private IEnumerable<Claim> CreateClaims(User user)
         {
             var userIdClaim = new Claim(JwtRegisteredClaimNames.Sub, user.Id.ToString());
-            var userEmailClaim = new Claim(JwtRegisteredClaimNames.Email, user.Email);
-            var userRoleClaim = new Claim("role", user.Role);
+            var userNameClaim = new Claim(ClaimTypes.Name, user.Id.ToString());
```
Comment (andrei-kruhliakou-itechart) [inline Airline_Web_API/Airline_Web_API/Services/JwtService.cs]: Why do you put user ID into user Name claim?

File: ClientApp/src/app/user-details/user-details.component.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  ngOnInit(): void {
+    this.userService.getUser().subscribe(user =>
+    {
+      this.user = user;
+      this.userForm = this.formBuilder.group({
+      firstName: new FormControl(this.user.firstName, [Validators.required, Validators.pattern(this.nameFormat)]),
+      lastName: new FormControl(this.user.lastName, [Validators.required, Validators.pattern(this.nameFormat)]),
+      email: new FormControl(this.user.email, [Validators.required, Validators.email]),
+      birthDate: new FormControl(this.user.birthDate, Validators.required)
+          });
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/user-details/user-details.component.ts]: Please fix indentation here

File: ClientApp/src/app/user-details/user-details.component.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  styleUrls: ['./user-details.component.css']
+})
+export class UserDetailsComponent implements OnInit {
+
+  user: User;
+  updateUser: UpdateUserModel;
+  userForm: FormGroup;
+  nameFormat = '[a-zA-Z\s]+$';
+  dateNow = new Date();
+  maxDate = new Date(new Date(this.dateNow).getTime() - 1000 * 60 * 60 * 24 * 365 * 14);
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/user-details/user-details.component.ts]: Shouldn't this expression '1000 * 60 * 60 * 24 * 365 * 14' be defined as constant with some meaningful name?
Maybe even a configuration setting? (Not required but think of it).

File: ClientApp/src/app/Services/user.service.ts [hunk 4]
```diff
@@ -24,27 +26,58 @@ export class UserService {
...
-  logout(): void {
+  logOut(): void {
     localStorage.removeItem('token');
   }
 
-  handleError(error: HttpErrorResponse): Observable<any>{
+  deleteUser(id: number): Observable<ResponseModel> {
+    const url = `${this.usersUrl}/${id}`;
+    return this.http.delete<ResponseModel>(url, this.httpOptions).pipe(
+      catchError(error => this.handleError(error)));
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/Services/user.service.ts]: Please place ); on a new line as you did in other pipe() calls above, be consistent with your code style.

File: Airline_Web_API/Airline_Web_API/Services/UserService.cs [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+            }
+
+            _mapper.Map(model, user);
+            await _context.SaveChangesAsync();
+
+             return new Response<string>
+             {
+                 Success = true,
+                 Message = "User was succesfully updated"
+             };
```
Comment (andrei-kruhliakou-itechart) [inline Airline_Web_API/Airline_Web_API/Services/UserService.cs]: Unnecessary indentation

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

File: Airline_Web_API/Airline_Web_API/Services/JwtService.cs (hunk 1 above)
Comment (ArinaSavchenko) [reply Airline_Web_API/Airline_Web_API/Services/JwtService.cs]: I've method in Startup file which checks if user is authorized and this methods requires this value
image

File: ClientApp/src/app/user-details/user-details.component.ts (hunk 2 above)
Comment (andrei-kruhliakou-itechart) [reply ClientApp/src/app/user-details/user-details.component.ts]: You've fixed it only partially

File: Airline_Web_API/Airline_Web_API/Services/JwtService.cs (hunk 1 above)
Comment (andrei-kruhliakou-itechart) [reply Airline_Web_API/Airline_Web_API/Services/JwtService.cs]: I mean you already have user ID in Sub claim, why have in the Name as well?

**Review by andrei-kruhliakou-itechart — APPROVED** [review]

File: Airline_Web_API/Airline_Web_API/Services/JwtService.cs (hunk 1 above)
Comment (ArinaSavchenko) [reply Airline_Web_API/Airline_Web_API/Services/JwtService.cs]: I corrected it. I didn't know how to do it, BUT now I've got it
image

**Pull request merged**